### ISSUE TITLE
REST API Bundle Product Details Price Issue

### ISSUE THREAD

**Issue by mack4hack** [magento/magento2#28431]
### Description
Bundle products have a price in the store and admin interface, but not in the Products Details REST API. 

### Preconditions
1. Magento 2.3.3

### Steps to reproduce
1. Create a bundle product and assign simple products to it.
![image](https://user-images.githubusercontent.com/11089409/83320211-f63e6380-a262-11ea-972a-220b9e91635b.png)

2. Enable 'Allow Anonymous Guest Access'  from Store -> Configuration -> Services -> Magento Web API

3. GET /rest/V1/products/{sku} to get a details of bundle product.

### Expected result
1. Children products must have price in response 

### Actual result
1. Price is 'null' of children products.
```
{
  "id": 1203,
  "sku": "Bundle Product",
  "name": "Bundle Product",
  "attribute_set_id": 4,
  "price": 0,
  "status": 1,
  "visibility": 4,
  "type_id": "bundle",
  "created_at": "2020-05-27 11:02:38",
  "updated_at": "2020-05-27 11:02:38",
  "weight": 0,
  "extension_attributes": {
    "website_ids": [
      1
    ],
    "bundle_product_options": [
      {
        "option_id": 1,
        "title": "option 1",
        "required": true,
        "type": "select",
        "position": 1,
        "sku": "Bundle Product",
        "product_links": [
          {
            "id": "1",
            "sku": "product_dynamic_3",
            "option_id": 1,
            "qty": 1,
            "position": 1,
            "is_default": false,
            "price": null, // Price is null
            "price_type": null,
            "can_change_quantity": 0
          },
          {
            "id": "2",
            "sku": "product_dynamic_4",
            "option_id": 1,
            "qty": 1,
            "position": 2,
            "is_default": false,
            "price": null, // Price is null
            "price_type": null,
            "can_change_quantity": 0
          }
        ]
      }
    ]
  },
  "product_links": [],
  "options": [],
  "media_gallery_entries": [],
  "tier_prices": [],
  ....
}
```

---

- [X] Severity: **S0**

**Comment by swathishs** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @swathishs, here is your [Magento instance](https://i-28431-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-28431-2-4-develop.instances.magento-community.engineering/admin_8912](https://i-28431-2-4-develop.instances.magento-community.engineering/admin_8912)
Login: `26be2afd` Password: `de534418de10`
Instance will be terminated in up to 3 hours.

**Comment by RanjitBShinde** [comment]
@magento give me 2.3.3 instance

**Comment by magento-engcom-team** [comment]
Hi @RanjitBShinde. Thank you for your request. I'm working on Magento 2.3.3 instance for you

**Comment by magento-engcom-team** [comment]
Hi @RanjitBShinde, here is your [Magento instance](https://i-28431-2-3-3.instances.magento-community.engineering).
Admin access: [https://i-28431-2-3-3.instances.magento-community.engineering/admin_3e16](https://i-28431-2-3-3.instances.magento-community.engineering/admin_3e16)
Login: `bcc08395` Password: `cb2c73841b16`
Instance will be terminated in up to 3 hours.

**Comment by RanjitBShinde** [comment]
@magento-engcom-team @magento I am able to replicate this issue, see following response of rest api

API End Point(GET) : https://i-28431-2-3-3.instances.magento-community.engineering/rest/V1/products/PRHC0002

Response:

{
    "id": 1201,
    "sku": "PRHC0002",
    "name": "Rest API bundle product",
    "attribute_set_id": 4,
    "price": 0,
    "status": 1,
    "visibility": 4,
    "type_id": "bundle",
    "created_at": "2020-06-01 06:13:21",
    "updated_at": "2020-06-01 06:14:24",
    "weight": 0,
    "extension_attributes": {
        "website_ids": [
            1
        ],
        "category_links": [
            {
                "position": 0,
                "category_id": "3"
            }
        ],
        "bundle_product_options": [
            {
                "option_id": 1,
                "title": "Test Bundle",
                "required": true,
                "type": "select",
                "position": 1,
                "sku": "PRHC0002",
                "product_links": [
                    {
                        "id": "1",
                        "sku": "product_dynamic_1",
                        "option_id": 1,
                        "qty": 1,
                        "position": 1,
                        "is_default": false,
                        "price": null,
                        "price_type": null,
                        "can_change_quantity": 0
                    },
                    {
                        "id": "2",
                        "sku": "product_dynamic_2",
                        "option_id": 1,
                        "qty": 1,
                        "position": 2,
                        "is_default": false,
                        "price": null,
                        "price_type": null,
                        "can_change_quantity": 0
                    },
                    {
                        "id": "3",
                        "sku": "product_dynamic_3",
                        "option_id": 1,
                        "qty": 1,
                        "position": 3,
                        "is_default": false,
                        "price": null,
                        "price_type": null,
                        "can_change_quantity": 0
                    }
                ]
            }
        ]
    },
    "product_links": [],
    "options": [],
    "media_gallery_entries": [],
    "tier_prices": [],
    "custom_attributes": [
        {
            "attribute_code": "shipment_type",
            "value": "0"
        },
        {
            "attribute_code": "options_container",
            "value": "container2"
        },
        {
            "attribute_code": "msrp_display_actual_price_type",
            "value": "0"
        },
        {
            "attribute_code": "url_key",
            "value": "rest-api-bundle-product"
        },
        {
            "attribute_code": "gift_message_available",
            "value": "2"
        },
        {
            "attribute_code": "price_view",
            "value": "0"
        },
        {
            "attribute_code": "required_options",
            "value": "1"
        },
        {
            "attribute_code": "has_options",
            "value": "1"
        },
        {
            "attribute_code": "meta_title",
            "value": "Rest API bundle product"
        },
        {
            "attribute_code": "sku_type",
            "value": "0"
        },
        {
            "attribute_code": "meta_keyword",
            "value": "Rest API bundle product"
        },
        {
            "attribute_code": "price_type",
            "value": "0"
        },
        {
            "attribute_code": "meta_description",
            "value": "Rest API bundle product "
        },
        {
            "attribute_code": "tax_class_id",
            "value": "2"
        },
        {
            "attribute_code": "weight_type",
            "value": "0"
        },
        {
            "attribute_code": "category_ids",
            "value": [
                "3"
            ]
        }
    ]
}

**Comment by RanjitBShinde** [comment]
@magento I am working on this

**Comment by RanjitBShinde** [comment]
@magento-engcom-team @magento I am able to replicate this issue, see following response of rest api

API End Point(GET) : https://i-28431-2-3-3.instances.magento-community.engineering/rest/V1/products/PRHC0002

Response:

{
"id": 1201,
"sku": "PRHC0002",
"name": "Rest API bundle product",
"attribute_set_id": 4,
"price": 0,
"status": 1,
"visibility": 4,
"type_id": "bundle",
"created_at": "2020-06-01 06:13:21",
"updated_at": "2020-06-01 06:14:24",
"weight": 0,
"extension_attributes": {
"website_ids": [
1
],
"category_links": [
{
"position": 0,
"category_id": "3"
}
],
"bundle_product_options": [
{
"option_id": 1,
"title": "Test Bundle",
"required": true,
"type": "select",
"position": 1,
"sku": "PRHC0002",
"product_links": [
{
"id": "1",
"sku": "product_dynamic_1",
"option_id": 1,
"qty": 1,
"position": 1,
"is_default": false,
"price": null,
"price_type": null,
"can_change_quantity": 0
},
{
"id": "2",
"sku": "product_dynamic_2",
"option_id": 1,
"qty": 1,
"position": 2,
"is_default": false,
"price": null,
"price_type": null,
"can_change_quantity": 0
},
{
"id": "3",
"sku": "product_dynamic_3",
"option_id": 1,
"qty": 1,
"position": 3,
"is_default": false,
"price": null,
"price_type": null,
"can_change_quantity": 0
}
]
}
]
},
"product_links": [],
"options": [],
"media_gallery_entries": [],
"tier_prices": [],
"custom_attributes": [
{
"attribute_code": "shipment_type",
"value": "0"
},
{
"attribute_code": "options_container",
"value": "container2"
},
{
"attribute_code": "msrp_display_actual_price_type",
"value": "0"
},
{
"attribute_code": "url_key",
"value": "rest-api-bundle-product"
},
{
"attribute_code": "gift_message_available",
"value": "2"
},
{
"attribute_code": "price_view",
"value": "0"
},
{
"attribute_code": "required_options",
"value": "1"
},
{
"attribute_code": "has_options",
"value": "1"
},
{
"attribute_code": "meta_title",
"value": "Rest API bundle product"
},
{
"attribute_code": "sku_type",
"value": "0"
},
{
"attribute_code": "meta_keyword",
"value": "Rest API bundle product"
},
{
"attribute_code": "price_type",
"value": "0"
},
{
"attribute_code": "meta_description",
"value": "Rest API bundle product "
},
{
"attribute_code": "tax_class_id",
"value": "2"
},
{
"attribute_code": "weight_type",
"value": "0"
},
{
"attribute_code": "category_ids",
"value": [
"3"
]
}
]
}

**Comment by RanjitBShinde** [comment]
@magento I am working on this

**Comment by RanjitBShinde** [comment]
Issue: Clear Description
Reproduced on 2.3.3
Issue: Confirmed
Component: Webapi
Component: Catalog

**Comment by mack4hack** [comment]
@swathishs please verify this issue.

**Comment by RanjitBShinde** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @RanjitBShinde. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by RanjitBShinde** [comment]
@magento issue is reproducible in magento 2.4-develop instance

**Comment by swathishs** [comment]
@mack4hack : The issue is reproducible in Magento 2.4 develop instance

Please find the attachment below:

1. Bundle product:
![image](https://user-images.githubusercontent.com/58288685/83518306-e9866d80-a4f7-11ea-882b-009e233253c4.png)

2. Api result:
{
    "id": 1201,
    "sku": "Testproduct",
    "name": "Testproduct",
    "attribute_set_id": 4,
    "price": 0,
    "status": 1,
    "visibility": 4,
    "type_id": "bundle",
    "created_at": "2020-06-02 11:55:25",
    "updated_at": "2020-06-02 11:55:25",
    "weight": 0,
    "extension_attributes": {
        "website_ids": [
            1
        ],
        "bundle_product_options": [
            {
                "option_id": 1,
                "title": "simple",
                "required": true,
                "type": "checkbox",
                "position": 1,
                "sku": "Testproduct",
                "product_links": [
                    {
                        "id": "1",
                        "sku": "product_dynamic_1",
                        "option_id": 1,
                        "qty": 10,
                        "position": 1,
                        "is_default": true,
                        "price": null,
                        "price_type": null,
                        "can_change_quantity": 0
                    },
                    {
                        "id": "2",
                        "sku": "product_dynamic_2",
                        "option_id": 1,
                        "qty": 10,
                        "position": 2,
                        "is_default": true,
                        "price": null,
                        "price_type": null,
                        "can_change_quantity": 0
                    },
                    {
                        "id": "3",
                        "sku": "product_dynamic_3",
                        "option_id": 1,
                        "qty": 10,
                        "position": 3,
                        "is_default": true,
                        "price": null,
                        "price_type": null,
                        "can_change_quantity": 0
                    },
                    {
                        "id": "4",
                        "sku": "product_dynamic_4",
                        "option_id": 1,
                        "qty": 10,
                        "position": 4,
                        "is_default": true,
                        "price": null,
                        "price_type": null,
                        "can_change_quantity": 0
                    }
                ]
            }
        ]
    },
    "product_links": [],
    "options": [],
    "media_gallery_entries": [],
    "tier_prices": [],
    "custom_attributes": [
        {
            "attribute_code": "shipment_type",
            "value": "0"
        },
        {
            "attribute_code": "options_container",
            "value": "container2"
        },
        {
            "attribute_code": "url_key",
            "value": "testproduct"
        },
        {
            "attribute_code": "msrp_display_actual_price_type",
            "value": "0"
        },
        {
            "attribute_code": "gift_message_available",
            "value": "2"
        },
        {
            "attribute_code": "price_view",
            "value": "0"
        },
        {
            "attribute_code": "required_options",
            "value": "1"
        },
        {
            "attribute_code": "has_options",
            "value": "1"
        },
        {
            "attribute_code": "meta_title",
            "value": "Testproduct"
        },
        {
            "attribute_code": "sku_type",
            "value": "0"
        },
        {
            "attribute_code": "meta_keyword",
            "value": "Testproduct"
        },
        {
            "attribute_code": "price_type",
            "value": "0"
        },
        {
            "attribute_code": "meta_description",
            "value": "Testproduct "
        },
        {
            "attribute_code": "tax_class_id",
            "value": "2"
        },
        {
            "attribute_code": "weight_type",
            "value": "0"
        },
        {
            "attribute_code": "category_ids",
            "value": []
        }
    ]
}

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @swathishs 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-34800` were created

**Issue Available**: @swathishs, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by syed-ziffity** [comment]
@magento I am working on this

**Comment by vbyno** [comment]
Isn't it an expected behavior for bundle products with dynamic prices? The price is not null only for fixed-price bundles, for dynamic ones the prices will be taken from the product at the moment order is created.

Otherwise, if we want the correct price of a simple product to be always displayed in product options of bundle product, we would need to:
1) introduce new `price_type` value (probably, 2)
2) update correspondent bundle prices each time simple product price is updated

**Comment by gabrieldagama** [comment]
Hi @mack4hack. Thank you for your report. 
The issue has been fixed in magento/magento2#28689 by @syed-ziffity in 2.4-develop branch
Related commit(s):
 - [8974e16d98b6154c64b5de6288d3d7ac7cc7291e](https://github.com/magento/magento2/commit/8974e16d98b6154c64b5de6288d3d7ac7cc7291e)
 - [e962d7c44e3d754ef8de4582d9e636918384b276](https://github.com/magento/magento2/commit/e962d7c44e3d754ef8de4582d9e636918384b276)
 - [2178617d083e9687f46df627ff4d913d63da99d7](https://github.com/magento/magento2/commit/2178617d083e9687f46df627ff4d913d63da99d7)
 - [7f4eccbb79f0a51f976d92c23c29254df8d2dde8](https://github.com/magento/magento2/commit/7f4eccbb79f0a51f976d92c23c29254df8d2dde8)
 - [34a64338cc046aaaf564bdd723f5719376a76b37](https://github.com/magento/magento2/commit/34a64338cc046aaaf564bdd723f5719376a76b37)
 - [1d82b50123a55291d95b488a60c1f7dfaffc9b95](https://github.com/magento/magento2/commit/1d82b50123a55291d95b488a60c1f7dfaffc9b95)

The fix will be available with the upcoming 2.4.2 release.
